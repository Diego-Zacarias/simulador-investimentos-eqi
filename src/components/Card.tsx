import React from "react";
import styled, { css } from "styled-components";

interface StyleProps {
  success?: boolean;
  testId?: string;
}

const Root = styled.div<StyleProps>`
  display: flex;
  width: 192px;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  background-color: #F4F4F4;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.5);
  h5 {
    margin: 2px 0 12px 0;
  }
  p {
    margin: 12px 0 2px 0;
    ${(props) =>
      props.success &&
      css`
        color: ${props => props.theme.colors.success};
        font-weight: bold;
      `};
  }
`;

interface CardProps {
  title: string;
  value: string;
  success?: boolean;
  testId?: string;
}
const Card = ({ title, value, success, testId }: CardProps) => {
  return (
    <Root success={success} testId={testId}>
      <h5>{title}</h5>
      <p>{value}</p>
    </Root>
  );
};

export default Card;
