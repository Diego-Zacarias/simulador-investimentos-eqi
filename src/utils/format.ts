export const format = (number: number): string => {
  return number.toLocaleString("pt-br", { style: "currency", currency: "BRL" });
};
