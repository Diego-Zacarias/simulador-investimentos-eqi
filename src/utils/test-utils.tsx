import React from "react";
import { render } from "@testing-library/react";
import ThemeProvider from "../styles/ThemeProvider";

const TestProvider = ({ children }: any): JSX.Element => (
      <ThemeProvider>{children}</ThemeProvider>
  );

export * from "@testing-library/react";

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: TestProvider, ...options });

export { customRender as render };
