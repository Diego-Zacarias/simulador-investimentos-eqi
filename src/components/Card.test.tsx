import { render, screen } from "../utils/test-utils";

import Card from "./Card";

test('Renderiza com o título "Título', () => {
  render(<Card title="Título" value="" />);
  const linkElement = screen.getByText(/Título/i);
  expect(linkElement).toBeInTheDocument();
});

test("Renderiza com a prop success true", () => {
  const { asFragment } = render(<Card title="" value="value" success={true} />);
  expect(asFragment()).toMatchSnapshot();
});
