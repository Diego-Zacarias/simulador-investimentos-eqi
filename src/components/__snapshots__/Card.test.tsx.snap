// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza com a prop success true 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 192px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #F4F4F4;
  box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.5);
}

.c0 h5 {
  margin: 2px 0 12px 0;
}

.c0 p {
  margin: 12px 0 2px 0;
  color: #469E2A;
  font-weight: bold;
}

<div
    class="c0"
  >
    <h5 />
    <p>
      value
    </p>
  </div>
</DocumentFragment>
`;
